st_aggrid/frontend/src/AgGrid.tsx

```diff
@@ -50,506 +50,517 @@ import "@fontsource/source-sans-pro"
+                this.api.exportDataAsExcel()
+            }
+        }
+    }
 
-      if (
-        this.props.args.excel_export_mode === "MULTIPLE_SHEETS" &&
-        this.props.args.ExcelExportMultipleSheetParams
-      ) {
-        let params = this.props.args.ExcelExportMultipleSheetParams
+    private handleExcelExport() {
+        if (this.props.args.excel_export_mode === "FILE_BLOB_IN_GRID_RESPONSE") {
+            let blob = this.api.getDataAsExcel() as Blob
+            let buffer;
+            (async () => {
+                await new Promise((resolve, reject) => {
+                    blob.arrayBuffer().then((v) => {
+                        buffer = encode(v)
+                        resolve(buffer)
+                    })
+                })
+            })()
+            return buffer
+        }
 
-        let data = params.data.map((v: string) =>
-          Buffer.from(decode(v)).toString("latin1")
-        )
-        params.data = data
+        if (this.props.args.excel_export_mode === "SHEET_BLOB_IN_GRID_RESPONSE") {
+            let blob = this.api.getSheetDataForExcel({
+                sheetName: Math.round(Date.now() / 1000).toString(),
+            })
+            if (blob) {
+                return encode(Buffer.from(blob, 'latin1')) ///Buffer.from(blob).toString('base64')
+            }
+        }
         
-        this.api.exportMultipleSheetsAsExcel(params)
-      }
-      if (this.props.args.excel_export_mode === "TRIGGER_DOWNLOAD") {
-        this.api.exportDataAsExcel()
+        return null
       }
-    }
-  }
-
-  private handleExcelExport() {
-    if (this.props.args.excel_export_mode === "FILE_BLOB_IN_GRID_RESPONSE") {
-      let blob = this.api.getDataAsExcel() as Blob
-      let buffer;
-      (async () => {
-        await new Promise((resolve, reject) => {
-          blob.arrayBuffer().then((v) => {
-            buffer = encode(v)
-            resolve(buffer)
-          })
-        })
-      })()
-      return buffer
-    }
 
-    if (this.props.args.excel_export_mode === "SHEET_BLOB_IN_GRID_RESPONSE") {
-      let blob = this.api.getSheetDataForExcel({
-        sheetName: Math.round(Date.now() / 1000).toString(),
-      })
-      if (blob) return encode(Buffer.from(blob, 'latin1')) ///Buffer.from(blob).toString('base64')
+    private resizeGridContainer() {
+        const renderedGridHeight = this.gridContainerRef.current?.clientHeight
+        if (renderedGridHeight && renderedGridHeight > 0 && renderedGridHeight != this.renderedGridHeightPrevious) {
+            this.renderedGridHeightPrevious = renderedGridHeight
+            Streamlit.setFrameHeight(renderedGridHeight)
+            // Run fitColumns only once when it first becomes visible with height > 0
+            // This should solve auto_size_mode issues with st.tabs
+            if (!this.fitColumnsDone) {
+                this.fitColumnsDone = true
+                this.fitColumns()
```

Comment:
Consider renaming to make if statement true on first run and false on remaining. Something like `fitOnFirstLoad` or similar.

Reply:
Renamed